projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -145,7 +145,7 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
         if (this.shouldGenerate) {
             this.setupColumns();
         }
-        this.notifyChanges(true);
+        this.cdr.markForCheck();
```

Comment:
This need to be changed in the tree-grid.component.ts, also.
Because now the issue is fixed for the grid, but it is still reproducible in the treeGrid